src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java

```diff
@@ -118,12 +125,16 @@ private void processAction(Action action) {
     }
   }
 
-  private ActionProcessingService getActionProcessingService(Action action) {
+  private ActionProcessingService getActionProcessingService(Action action) throws CTPException {
```

Comment:
rather than declaring `throws CTPException` I'd prefer it if we caught the exception and threw an `IllegalStateException` which is a runtime exception, so it doesn't need to be 'checked'. It's a few more lines of code, but it's a step towards getting rid of checked exceptions